carla/recourse_methods/catalog/clue/library/clue_ml/src/utils.py

```diff
@@ -0,0 +1,320 @@
+from __future__ import division, print_function
+
+import os
+import sys
+
+import torch
+import torch.nn as nn
+from torch.autograd import Variable
+
+try:
+    import cPickle as pickle
+except:
+    import pickle
+
+
+def mkdir(paths):
+    if not isinstance(paths, (list, tuple)):
+        paths = [paths]
+    for path in paths:
+        if not os.path.isdir(path):
+            os.makedirs(path, mode=0o777)
+
+
+import torch.nn as nn
+
+suffixes = ["B", "KB", "MB", "GB", "TB", "PB"]
+
+
+def humansize(nbytes):
+    i = 0
+    while nbytes >= 1024 and i < len(suffixes) - 1:
+        nbytes /= 1024.0
+        i += 1
+    f = "%.2f" % nbytes
+    return "%s%s" % (f, suffixes[i])
+
+
+def get_num_batches(nb_samples, batch_size, roundup=True):
+    if roundup:
+        return (
+            nb_samples + (-nb_samples % batch_size)
+        ) / batch_size  # roundup division
+    else:
+        return nb_samples / batch_size
+
+
+def generate_ind_batch(nb_samples, batch_size, random=True, roundup=True):
+    if random:
+        ind = np.random.permutation(nb_samples)
+    else:
+        ind = range(int(nb_samples))
+    for i in range(int(get_num_batches(nb_samples, batch_size, roundup))):
+        yield ind[i * batch_size : (i + 1) * batch_size]
+
+
+def to_variable(var=(), cuda=False, volatile=False):
+    out = []
+    for v in var:
+        if isinstance(v, np.ndarray):
+            v = torch.from_numpy(v).type(torch.FloatTensor)
+        if not v.is_cuda and cuda:
+            v = v.cuda()
+        if not isinstance(v, Variable):
+            v = Variable(v, volatile=volatile)
+        out.append(v)
+    return out
+
+
+def cprint(color, text, **kwargs):
```

Comment:
If we want to color print stuff lets do it on the package level, not here in the library